extensions/job_management/standard/requirements/job-management/start/REQ_response.adoc

```diff
@@ -0,0 +1,8 @@
+[[req_job-management_start_response]]
+[requirement]
+====
+[%metadata]
+label:: /req/job-management/start-response
+part:: A successful execution of the operation SHALL be reported as a response with a HTTP status code '200'.
```

Comment:
Using `POST /jobs/{jobId}/results`, the server should probably still have the option to negotiate the execution mode. In other words, `sync` and 200 would be used by default, but a `Prefer: respond-async` would allow the server to trigger this job, although it would not respond with results right away. If `async` is selected this way, the response would instead the usual Job Status with monitoring. Also, a 202 would have to be used, since no "job" is created in that case. Once completed, the results can be retrieved from `GET /jobs/{jobId}/results`.

Reply:
Would you not expect the sync result already from POST /jobs?

Reply:
You could do either of the following: 

1. `POST /jobs` + `Prefer: wait=X` to respond results inline (no need for other request)

2. `POST /jobs` + `Prefer: respond-async` responds with Job Status
   1. starts the job whenever resources are ready
   2. `GET /jobs/{jobId}/results` to retrieve results once `status: succeeded`
   
3. `POST /jobs` + `Prefer: wait=X` + `status: create`
   1. places the job in `created` state, until triggered later on
   2. `POST /jobs/{jobId}/results` to trigger the job and return the results inline

4. `POST /jobs` + `Prefer: respond-async` + `status: create`
   1. places the job in `created` state, until triggered later on
   2. `POST /jobs/{jobId}/results` to trigger the job and return with Job Status
   3. `GET /jobs/{jobId}/results` to retrieve results once `status: succeeded`

Here, the `202` would make sense for 4.ii, whereas `200` for 3.ii

Reply:
I feel like you have too many options in OAP. You are implementing things server side that usually a client would handle. Then the server (and client) implementations are much simpler. openEO API is much simpler, but can still do everything you do here through their clients.

Reply:
Well, we've had users expecting all these combinations so far, so we must support them to accommodate everyone. If Part 4 only allows working with openEO style jobs, it's not much of an extension for OAP.

Reply:
The first 2 are a direct mapping of what `POST /processes/{processId}/execution` defines. Those have to be mapped as is on `POST /jobs` to allow a server/client to work only with the `/jobs` endpoint and integrating workflows.

The other 2 are for openEO alignment, allowing the creation of a job first, and then triggering its execution with a second request. At the moment in OAP, there is no such concept of job creation not queued right away, so the first 2 points needs to be available for servers that do not intend to create jobs this way, but still want to define workflows. 

What you're asking for is to consider only openEO's perspective, which of course is easier, because you're thinking only about your use cases.

And to clarify, the "users" here that I was referring to are the participants of TB20-GDC. We have had discussions trying to show others (including you) alternatives, but nobody wants to concede anything. So, we are in this situation where all 3 modes of sync, async, and create/trigger execution are supported by `POST /jobs` as alternatives of running processes by different users. 

The best simplification that can be done is by merging (3) and (4), basically not allowing switching `Prefer` when doing `POST /jobs/{jobId}/results`. It would have to be configured from the get go when doing `POST /jobs` or by follow-up `PATCH /jobs/{jobId}` to modify it.

Reply:
> The first 2 are a direct mapping of what `POST /processes/{processId}/execution` defines. Those have to be mapped as is on `POST /jobs` to allow a server/client to work only with the `/jobs` endpoint and integrating workflows.

Why if there's already an alternative?

> The other 2 are for openEO alignment, allowing the creation of a job first, and then triggering its execution with a second request. At the moment in OAP, there is no such concept of job creation not queued right away, so the first 2 points needs to be available for servers that do not intend to create jobs this way, but still want to define workflows.

3 and 4 are not available in openEO right now?! 4 would be if you remove the "and return with Job Status". 
I still think not all of this needs to be available in favor of clients that can make this happen in one call if you want.
But I understand that historically OGC was about servers, not clients, which I think is a massiv mistake.

> What you're asking for is to consider only openEO's perspective, which of course is easier, because you're thinking only about your use cases.

No.

> And to clarify, the "users" here that I was referring to are the participants of TB20-GDC. We have had discussions trying to show others (including you) alternatives, but nobody wants to concede anything. So, we are in this situation where all 3 modes of sync, async, and create/trigger execution are supported by `POST /jobs` as alternatives of running processes by different users.

No. We had discussions where each side was making compromises and we were on a good way, but it seems we are not following the first meeting where I was still available. I guess we should listen to the first meeting recording again. The thing is, if no one make compromises we should NOT align, because then we end up with an overengineered complex API. Then it's easier to have two separate ones. I was under the impression that we were on a good way, but this PR is not reflecting this as far as I can tell... 

> The best simplification that can be done is by merging (3) and (4), basically not allowing switching `Prefer` when doing `POST /jobs/{jobId}/results`. It would have to be configured from the get go when doing `POST /jobs` or by follow-up `PATCH /jobs/{jobId}` to modify it.

I think (2) and (3) can be easily be removed. You can simply solve 2 with two subsequent requests and why should there be two ways to process synchronously? We need to simplify not allow everything to be solved in 100 ways.

Reply:
> Why if there's already an alternative?

It is not a direct alternative, because `/processes/{processId}/execution` does not allow *all* capabilities that `/jobs` offers. Notably, defining workflow/graphs that do not have only one top-level process, and providing additional job endpoints for metadata.

> 3 and 4 are not available in openEO right now?! 

You misunderstood. It's the other way around. It is not available in OAP! 
Only openEO defines such capability to create a job without putting it in queue. In OAP, when you POST the job, it is either executed in sync or async, but it is done right away. OAP does not allow creating a job, modify it, and submit it in queue after. This is added by Part 4.

>  think (2) and (3) can be easily be removed. You can simply solve 2 with two subsequent requests and why should there be two ways to process synchronously?

I agree about removing (3), and keeping only (4) for the openEO-style execution.

However, (2) should remain. It is completely different from (3)/(4) use case, because it locks the job right away and places it in queue (just like if it was executed in sync right away). Why force the client to do a subsequent request to trigger the job, when this hint can be supplied immediately via the `Prefer` header? In most situations, clients that submit a job intend for it to run right away, because their goal is to obtain the result from it. Avoiding an extra request for every single time a job is submitted effectively cuts the network traffic by 2. If the client intends to do more with the job (modify it, review it, whatever) before executing it, it makes more sense IMO that this detail should be specified explicitly (ie: `status: create`). Then, the client/server agree explicitly that a subsequent request is needed for the actual execution trigger.